mqtt/mqtt-bridge/src/persist/waking_state/waking_store.rs

```diff
@@ -0,0 +1,109 @@
+#![allow(dead_code)] // TODO remove when ready
+
+use std::{collections::hash_map::Entry, collections::HashMap, task::Waker};
+
+use bincode::{self};
+use mqtt3::proto::Publication;
+use rocksdb::{IteratorMode, Options, DB};
+use uuid::Uuid;
+
+use crate::persist::{waking_state::StreamWakeableState, Key, PersistError};
+
+/// When elements are retrieved they are added to the in flight collection, but kept in the original db store.
+/// Only when elements are removed from the in-flight collection they will be removed from the store.
+pub struct WakingStore {
+    db: DB,
+    in_flight: HashMap<Key, Publication>,
+    waker: Option<Waker>,
+    column_family: String,
+}
+
+impl WakingStore {
+    pub fn new(mut db: DB) -> Result<Self, PersistError> {
+        let column_family = Uuid::new_v4().to_string();
+        db.create_cf(column_family.clone(), &Options::default())
+            .map_err(PersistError::CreateColumnFamily)?;
+
+        Ok(Self {
+            db,
+            in_flight: HashMap::new(),
+            waker: None,
+            column_family,
+        })
+    }
+}
+
+impl StreamWakeableState for WakingStore {
+    fn insert(&mut self, key: Key, value: Publication) -> Result<(), PersistError> {
+        let key_bytes = bincode::serialize(&key).map_err(PersistError::Serialization)?;
+        let publication_bytes = bincode::serialize(&value).map_err(PersistError::Serialization)?;
+
+        let column_family = self
```

Comment:
getting a column family handle is the cheap operation?

Reply:
lets extract getting a ColumnFamily into private method?

Reply:
I assumed it is. Here is the documentation:
https://docs.rs/rocksdb/0.15.0/rocksdb/struct.DB.html#method.cf_handle